drivers/platform/xilinx/xilinx_spi_pl.c

```diff
@@ -326,10 +335,81 @@ static int32_t xil_spi_write_and_read_pl(struct spi_desc *desc, uint8_t *data,
 	return SUCCESS;
 }
 
+/* Read/write FIFOs to/from buffers and update it structure accordingly */
+static void inline _transfer_and_update(struct xspi_desc *xdesc,
+					struct msg_iterator *it)
+{
+	uint32_t		i;
+	uint32_t		len;
+
+	do {
+		len = it->msg->bytes_number - it->i;
+		if (it->is_write)
+			i = _wr_fifo(xdesc, it->buff + it->i, len);
+		else if (it->buff)
+			i = _rd_fifo(xdesc, it->buff + it->i, len);
+		else
+			/* Upper layer don't provided a read buffer */
+			i = _rd_fifo(xdesc, NULL, len);
+
+		it->i += i;
+		/* If current msg have been sent/read go to next one */
+		if (it->i == it->msg->bytes_number) {
+			it->msg = it->msg->next;
+			it->i = 0;
+			if (it->is_write)
+				it->buff = it->msg->tx_buff;
+			else
+				it->buff = it->msg->rx_buff;
+		}
+	} while (i == len && it->msg); /* Until FIFO is full or empty */
+}
+
+/* SPI polled transfer of multiple messages */
+static int32_t xil_spi_write_and_read_list_pl(struct spi_desc *desc,
+				struct spi_msg_list *head)
+{
+	struct xspi_desc	*xdesc;
+	struct msg_iterator	rx;
+	struct msg_iterator	tx;
+
+	if (!desc || !desc->extra)
+		return -EINVAL;
+
+	xdesc = desc->extra;
+	_update_mode(desc);
+
+	/* Initialize rx and tx structures */
+	rx.msg = head;
+	rx.buff = head->rx_buff;
+	rx.i = 0;
+	rx.is_write = 0;
+
+	tx.msg = head;
+	tx.buff = head->tx_buff;
+	tx.i = 0;
+	tx.is_write = 1;
+
+	xil_spi_cs_assert(desc, 1);
```

Comment:
This allows multiple transfers with CS asserted but in my opinion is not flexible enough. Like on linux, and like we discussed in the meeting, there should be a possibility to assert/deassert CS at the beggining / end of each transfer since we are at it.